### PR TITLE
fix(pubsub): Reintroduces 'name' field to google pubsub publishers

### PR DESCRIPTION
**Pull request by ttomsu**
part of https://github.com/spinnaker/spinnaker/issues/2784

We used to have a operator-defined `name` field, but I took it out in the last revamp. Adding it back in to support the (probably?) niche use case of the same topic name in different projects.

This means that in order to trigger notifications, we match on `publisherName` instead of `topic` in the incoming notification map.

### CODE REVIEW

File: echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 package com.netflix.spinnaker.echo.pubsub;
 
+import java.util.List;
```
Comment (ezimanyi) [inline echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java]: I think normally the `java.util.*` goes at the end in our repos

File: echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java [hunk 2]
```diff
@@ -57,4 +60,27 @@
...
+        .pubsubSystem(p.getPubsubSystem().toString())
+        .publisherName(p.getName())
+        .topicName(p.getTopicName())
+        .build())
+      .collect(Collectors.toList());
+  }
+
+  @Data
+  @Builder
+  @NoArgsConstructor
```
Comment (ezimanyi) [inline echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java]: Minor point; I think that `@Builder` implies `@AllArgsConstructor` (which is what it uses) so we could remove that and also remove the `@NoArgsConstructor`, though I think in practice we tend to add all these annotations to data classes.

File: echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java (hunk 1 above)
Comment (ttomsu) [reply echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java]: Boo intellij formatter...

File: echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java (hunk 2 above)
Comment (ttomsu) [reply echo-pubsub-core/src/main/java/com/netflix/spinnaker/echo/pubsub/PubsubSubscriptionController.java]: Removed both.

**Comment by ttomsu** [comment]
1. Nobody should be using this yet - if there were, they probably would have complained when I removed it back in December.

2. Additional changes to Orca and Deck are linked in https://github.com/spinnaker/spinnaker/issues/2784

**Pull request merged**